recipes/qpdf/all/patches/0002-libqpdf-cmake-deps-jpeg-zlib-openssl.patch

```diff
@@ -0,0 +1,107 @@
+diff --git a/libqpdf/CMakeLists.txt b/libqpdf/CMakeLists.txt
+index 7053e205..7bfcc5fc 100644
+--- a/libqpdf/CMakeLists.txt
++++ b/libqpdf/CMakeLists.txt
+@@ -128,7 +128,6 @@ include(CheckSymbolExists)
+ set(dep_include_directories)
+ set(dep_link_directories)
+ set(dep_link_libraries)
+-set(ANYTHING_MISSING 0)
+ 
+ if(WIN32 AND (EXISTS ${qpdf_SOURCE_DIR}/external-libs))
+   set(EXTERNAL_LIBS 1)
+@@ -161,9 +160,6 @@ if(NOT EXTERNAL_LIBS)
+     if(ZLIB_H_PATH AND ZLIB_LIB_PATH)
+       list(APPEND dep_include_directories ${ZLIB_H_PATH})
+       list(APPEND dep_link_libraries ${ZLIB_LIB_PATH})
+-    else()
+-      message(SEND_ERROR "zlib not found")
+-      set(ANYTHING_MISSING 1)
+     endif()
+   endif()
+ endif()
+@@ -182,9 +178,6 @@ if(NOT EXTERNAL_LIBS)
+       list(APPEND dep_include_directories ${LIBJPEG_H_PATH})
+       list(APPEND dep_link_libraries ${LIBJPEG_LIB_PATH})
+       set(JPEG_INCLUDE ${LIBJPEG_H_PATH})
+-    else()
+-      message(SEND_ERROR "libjpeg not found")
+-      set(ANYTHING_MISSING 1)
+     endif()
+   endif()
+ endif()
+@@ -220,9 +213,6 @@ if(USE_IMPLICIT_CRYPTO OR REQUIRE_CRYPTO_OPENSSL)
+         list(APPEND dep_link_libraries ${OPENSSL_LIB_PATH})
+         set(USE_CRYPTO_OPENSSL ON)
+         set(FOUND_CRYPTO ON)
+-      elseif(REQUIRE_CRYPTO_OPENSSL)
+-        message(SEND_ERROR "openssl not found")
+-        set(ANYTHING_MISSING 1)
+       endif()
+     endif()
+   endif()
+@@ -241,9 +231,6 @@ if(USE_IMPLICIT_CRYPTO OR REQUIRE_CRYPTO_GNUTLS)
+       list(APPEND dep_link_libraries ${GNUTLS_LIB_PATH})
+       set(USE_CRYPTO_GNUTLS ON)
+       set(FOUND_CRYPTO ON)
+-    elseif(REQUIRE_CRYPTO_GNUTLS)
+-      message(SEND_ERROR "gnutls not found")
+-      set(ANYTHING_MISSING 1)
+     endif()
+   endif()
+ endif()
+@@ -268,14 +255,9 @@ if(FOUND_CRYPTO)
+       set(DEFAULT_CRYPTO "native")
+     endif()
+   endif()
+-else()
+-  message(SEND_ERROR "no crypto provider is available")
+-  set(ANYTHING_MISSING 1)
+-endif()
+-if(ANYTHING_MISSING)
+-  message(FATAL_ERROR "Missing dependencies; unable to continue")
+ endif()
+ 
++
+ message(STATUS "")
+ message(STATUS "*** Crypto Summary ***")
+ message(STATUS "  GNU TLS crypto enabled: " ${USE_CRYPTO_GNUTLS})
+@@ -403,6 +385,11 @@ endif()
+ # use PIC for the object library so we don't have to compile twice.
+ set(OBJECT_LIB libqpdf_object)
+ add_library(${OBJECT_LIB} OBJECT ${libqpdf_SOURCES})
++find_package(JPEG REQUIRED)
++find_package(ZLIB REQUIRED)
++find_package(OpenSSL REQUIRED)
++target_link_libraries(${OBJECT_LIB} PUBLIC JPEG::JPEG ZLIB::ZLIB OpenSSL::SSL)
```

Comment:
I am a little surprised it's not using `OpenSSL::Crypto` given the name of the option

Reply:
I was going for crypto as wording, since i picked it up from the qpdf dependency docu:
https://qpdf.readthedocs.io/en/stable/installation.html#basic-dependencies

Nevertheless, when there is a more meaningful crypto dependency management naming scheme in other packages, it would be better to sync them (e.g like with_jpeg in severral packages.) When i search through other packages, i think with_ssl is way more often used than with_crypto. So i will change it accordingly.